docs/guides/deployment/docker.md

```diff
@@ -12,129 +12,227 @@ search:
+
+## Writing a Dockerfile
+
+First let's write a directory to work from, `prefect-docker-guide`.
+
+<div class="terminal">
+```bash
+mkdir prefect-docker-guide
+cd prefect-docker-guide
+```
+</div>
+
+In this directory, we'll create a sub-directory named `flows` and put our flow script from the [Deployments](/tutorial/deployments/) tutorial in it.
 
-- We'll use the flow script and deployment from the [Deployments](/tutorial/deployments/) tutorial. 
-- You must run a standalone Prefect server (`prefect server start`) or use Prefect Cloud.
-- You'll need [Docker Engine](https://docs.docker.com/engine/) installed and running on the same machine as your agent.
+<div class="terminal">
+```bash
+mkdir flows
+cd flows
+touch prefect-docker-guide-flow.py
+```
+</div>
 
-[Docker Desktop](https://www.docker.com/products/docker-desktop) works fine for local testing if you don't already have Docker Engine configured in your environment.
+Here's the flow code for reference:
 
-!!! note "Run a Prefect server"
-    This guide assumes you're already running a Prefect server with `prefect server start`, as described in the [Deployments](/tutorial/deployments/) tutorial.
-    
-    If you shut down the server, you can start it again by opening another terminal session and starting the Prefect server with the `prefect server start` CLI command.
+```python title="prefect-docker-guide-flow.py"
+import httpx
+from prefect import flow
 
-## Storing Prefect Flow Code in a Docker Image 
 
-First let's create a directory to work from, `docker-tutorial`.
+@flow(log_prints=True)
+def get_repo_info(repo_name: str = "PrefectHQ/prefect"):
+    url = f"https://api.github.com/repos/{repo_name}"
+    response = httpx.get(url)
+    response.raise_for_status()
+    repo = response.json()
+    print(f"{repo_name} repository statistics 🤓:")
+    print(f"Stars 🌠 : {repo['stargazers_count']}")
+    print(f"Forks 🍴 : {repo['forks_count']}")
 
-In this directory, you will create a sub-directory named `flows` and put your flow script from the [Deployments](/tutorial/deployments/) tutorial. In this case, I've named the flow `docker-tutorial-flow.py`.
 
-The next file you will add to the `docker-tutorial` directory is a `requirements.txt`.  In this file make sure to include all dependencies that are required for your `docker-tutorial-flow.py` script.  
+if __name__ == "__main__":
+    get_repo_info.serve(name="prefect-docker-guide")
+```
 
-Next, you will create a `Dockerfile` that will be used to create a Docker image that will also store the flow code.  This `Dockerfile` should look similar to the following:  
+The next file we'll add to the `prefect-docker-guide` directory is a `requirements.txt`. We'll include all dependencies required for our `prefect-docker-guide-flow.py` script in the Docker image we'll build.
 
+<div class="terminal">
 ```bash
+touch requirements.txt
+```
+</div>
+
+Here's what we'll put in our `requirements.txt` file:
+
+```txt title="requirements.txt"
+httpx
```

Comment:
I recommend adding `prefect>=2.12.0` to this list for portability across other Docker files